server/src/main/java/org/candlepin/async/JobDataConverter.java

```diff
@@ -0,0 +1,358 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async;
+
+import org.candlepin.hibernate.AbstractJsonConverter;
+import org.candlepin.model.AsyncJobStatus.SerializedJobData;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+
+
+public class JobDataConverter extends AbstractJsonConverter<SerializedJobData> {
+    private static Logger log = LoggerFactory.getLogger(JobDataConverter.class);
+
+    private static final String METADATA_FIELD_NAME = "metadata";
+    private static final String JOB_ARGUMENTS_FIELD_NAME = "job_arguments";
+    private static final String JOB_CONSTRAINTS_FIELD_NAME = "job_constraints";
+    private static final String JOB_OUTPUT_FIELD_NAME = "job_output";
```

Comment:
I'm generally opposed to job outputs having to be serialized/deserialized. Once a job is finished and its output is persisted, there is no more processing to be done on it (it is simply retrieved and sent to the client); isn't it a static piece of data whose structure we don't care from then on?

Reply:
Yeah, probably. There's also some standardization that we could do with job output. With the exception of ImportJob and ExportJob, we don't use the output for much more than a sort of logging output to be sent with the job status. We could just as easily change the signature to be void and do what we used to do: setJobResult (though this has its own flow problems). At this point, we could apply the same type restrictions as everything else, serialize it on assignment, and then never touch it again.